base/system/smss/pagefile.c

```diff
@@ -20,8 +20,26 @@
 //
 #define STANDARD_PAGING_FILE_NAME       L"\\??\\?:\\pagefile.sys"
 #define STANDARD_DRIVE_LETTER_OFFSET    4
-#define MEGABYTE                        0x100000UL
-#define MAXIMUM_PAGEFILE_SIZE           (4095 * MEGABYTE)
+#define MAX_PAGING_FILES                16  // See also in ntoskrnl/include/internal/mm.h
+#define MEGABYTE                        (1024 * 1024)
+
+// #define MINIMUM_PAGEFILE_SIZE           (256 * PAGE_SIZE) // 1 MB
+
+/* Maximum pagefile sizes for different architectures */
+#if defined(_M_IX86)
+#define MAXIMUM_PAGEFILE_SIZE           (4095ULL * MEGABYTE) // 4095 MB
+/* PAE uses the same size as x64 */
+#define MAXIMUM_PAGEFILE_SIZE_PAE       (16ULL * 1024 * 1024 * MEGABYTE) // 16 TB
+#elif defined (_M_X64)
+#define MAXIMUM_PAGEFILE_SIZE           (16ULL * 1024 * 1024 * MEGABYTE) // 16 TB
+#elif defined (_M_IA64)
+#define MAXIMUM_PAGEFILE_SIZE           (32ULL * 1024 * 1024 * MEGABYTE) // 32 TB
+#else
+/* On unknown architectures, default to 4095 MB */
+#pragma message("Unknown architecture")
+#define MAXIMUM_PAGEFILE_SIZE           (4095ULL * MEGABYTE)
```

Comment:
I have some NT ARM kernels if needed :slightly_smiling_face:

Reply:
I would be interested in these (ARM32? ARM64?)
Also, related to these, do you know whether `_M_X86` is defined for both regular x86 and ARM32 (aka. "any" 32-bit platform) or not?

Reply:
There are `_M_ARM` and `_M_ARM64` accordingly. And `_WIN64` is defined only for latter.

Reply:
Here are my results for the maximum sizes
(If you think about an explanation why they seem to change between Windows 8, 8.1 and 10, please let me know):

NT 5.2 or 6.1 x86
- SMSS: 0xFFF00000 (== 4095 * 1024*1024)
- NTOS: 0xFFFFF000 (== (1 * 1024*1024-1) * 4096)

NT 5.2 or 6.1 x64:
- SMSS: 0x1000`00000000 (== 16 * 1024*1024*1024*1024)
- NTOS: 0x0FFF`FFFFF000 (== (4 * 1024*1024*1024-1) * 4096)

--------

NT 6.2 ARM v7
- SMSS: 0xFFF00000
- NTOS: 0xFFFFF000
(observe: different values).
Note: Same values as NT 5.2 or 6.1 x86.

NT 6.3 ARM v7
- SMSS: 0x7FFFF000 (== (512*1024-1) * 4096)
- NTOS: 0x7FFFF000
(here they agreed)

NT 10.x ARMv7
- SMSS: 0x7FFFF000
- NTOS: 0x7FFFE000 (== (512*1024-2) * 4096)
(observe: different values)

NT 10.x ARM64
- SMSS: 0x0FFF`FFFFF000 (== (4 * 1024*1024*1024-1) * 4096)  <-- as in NT 5.x or 6.x x64.
- NTOS: 0x0FFF`FFFFE000 (== (4 * 1024*1024*1024-2) * 4096)
(observe: different values)
Note: "minus 2" instead of "minus 1" (as in x64) gives the observed value.